chain/chain/src/chain.rs

```diff
@@ -391,6 +396,60 @@ pub fn check_known(
     check_known_store(chain, block_hash)
 }
 
+// Cache with the mappint from CryptoHash (blocks, chunks) to the number milliseconds that it took to process them.
+// Used only for debugging purposes.
+pub static CRYPTO_HASH_TIMER_RESULTS: Lazy<Mutex<LruCache<CryptoHash, u64>>> =
+    Lazy::new(|| Mutex::new(LruCache::new(10000)));
+
+/// Struct to measure computation times related to different CryptoHashes (for example chunk or block computations).
+/// It stores the data in the global LRU cache, which allows it to be read afterwards.
+///
+/// Example use:
+/// ```rust, ignore
+/// {
+///     let _timer = CryptoHashTime(block.hash);
+///     // Do stuff with the block
+///     // timer gets released at the end of this block
+/// }
+/// ```
+pub struct CryptoHashTimer {
+    key: CryptoHash,
+    start: Instant,
+}
+
+impl CryptoHashTimer {
+    pub fn new(key: CryptoHash) -> Self {
+        CryptoHashTimer { key, start: Clock::instant() }
+    }
+    pub fn get_timer_value(key: CryptoHash) -> Option<u64> {
+        CRYPTO_HASH_TIMER_RESULTS.lock().unwrap().get(&key).cloned()
+    }
+}
+
+impl Drop for CryptoHashTimer {
+    fn drop(&mut self) {
+        CRYPTO_HASH_TIMER_RESULTS
+            .lock()
+            .unwrap()
+            .put(self.key, (Clock::instant() - self.start).as_millis() as u64);
+    }
+}
+
+#[test]
+fn test_crypto_hash_timer() {
+    let crypto_hash = CryptoHash::from_str("s3N6V7CNAN2Eg6vfivMVHR4hbMZeh72fTmYbrC6dXBT").unwrap();
```

Comment:
```suggestion
    let crypto_hash: CryptoHash = "s3N6V7CNAN2Eg6vfivMVHR4hbMZeh72fTmYbrC6dXBT").parse().unwrap();
```

Rule of thumb: only import `FromStr` if you want to write an `impl`. The interface for *using* this trait is `str::parse` method.

Reply:
For the future: rather than gating imports with `#[cfg(test)] use ...` use the following pattern instead:

```rust
#[cfg(test)]
mod tests {
  // Bring everything from the parent module into the scope
  use super::*;

  // Any additional imports now don't need to be gated
  use foo::bar::baz;

  #[test]
  fn test_crypto_hash_timer() {
  }
}
```

If you really have just a single test function, the following would also work:

```rust
#[test]
fn test_crypto_hash_timer() {
  use foo::bar::baz;
}
```